CONTRIBUTING.md

```diff
@@ -23,34 +23,62 @@ If you find a bug anywhere, you can fix it.
+
+2. Clone your forked repository, this will download your copy of repository in your computer. To do this, open your terminal (command prompt/bash/git bash) and enter the following command, paste your link after the word **clone** without the **<>**.
 ``` 
-git clone https://github.com/<your-username>/website-welcome/ 
+git clone <link which you copied in the step 1>
 ```
 
-3. Add the original repository as a remote repository.
+3. Add the original repository as a remote repository, so that you can anytime pull the latest changes from the main repository which is being deployed, this needs to be done only for the first time.
 ``` 
-git remote add https://github.com/Real-Dev-Squad/website-welcome/ 
+git remote add upstream https://github.com/Real-Dev-Squad/website-welcome/ 
+```
+To make sure you always have the latest copy of the main repository before starting with your changes, execute the following command:
+```
+git pull upstream main
 ```
 
-4. Create a new branch to work on.
+4. Create a new branch to work on. We require a different branch so that we always have a stable, working version in the default (main) branch.
 ``` 
 git checkout -b <branch-name> 
 ```
 
-5. Do your changes.
+For example, if I want to name my branch as **develop**, I'll enter the following command:
+``` 
+git checkout -b develop
+```
+
+5. Perform the tasks you wanted to, can be anything, ranging from fixing simple typo to re-designing the whole page!
 
-6. Commit them.
+6. Now you have made the changes, though they are saved in your system, Git doesn't know what changes you've done. So you have **commit** your changes. First step is to add the files which you want to add to the staging area, the dot after **add** in the first command tells Git to check for changes in all the files. The second step is about committing your changes. The message part is short description of your commit, like "fixed the css for homepage".
 ``` 
 git add .
 git commit -m "Write message about your commit" 
 ```
 
-7. Push them to the original repository.
+7. Now your branch (develop, in my case) has the changes you made, once you're sure that your code is stable, it's time to merge the changes into the main branch of your local machine. The first command takes you to the main branch and the second command helps merge the other branch with the main branch.
 ``` 
-git push origin 
+git checkout main
+git merge <branch-name>
+```
+
+For example, if the branch name is develop:
 ```
+git merge develop
+```
+
+8. Now it is the time to push the changes of your local main branch to your copy of the repository (aka forked repository) which is on Github (which we cloned in step 2).
+```
+git push origin
+```
+
+9. Now go to your forked repository and you'll see that **This branch is xx commits ahead of Real-Dev-Squad:main.**  Right next to it will be an option to **Pull Request**. Click on it, submit your pull request (also known as *PR*) explaining what you've done.
```

Comment:
Should we add here--- make sure that you have the latest changes from the original repo.